PeerioTechnologies/peerio-desktop
# feat: don't allow more than one instance of app

In production, if another instance is launched, it will detect that the
app is already running, and quit. The first instance will bring itself
to the front.

To turn this off, pass `--allow-multiple-instances` command-line argument
(on macOS, the app needs to be launched from command-line for that).

^ this flag is for @ebilg :)


#### Relevant info and issue/PR links 
 
#### Testing instructions  

Testing needs to be performed with production build (`npm run internal-build`) on all operating systems.

1. Launch the app.
2. Launch it again.
3. The second instance should quit before it appears and the first one should re-appear and become focused.
4. Minimize the app.
5. Launch again.
6. The second instance should quit before it appears and the first one should unminimize itself and become focused.

Please ensure that the first instance does bring itself to the front on Windows, I'm a bit concerned about it.

If you want to test `--allow-multiple-instances` note that it needs to be passed to the very first instance and every additional instances to launch.

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
